server/neptune/workflows/internal/deploy/terraform/runner.go

```diff
@@ -14,24 +18,42 @@ type stateReceiver interface {
 	Receive(ctx workflow.Context, c workflow.ReceiveChannel, deploymentInfo DeploymentInfo)
 }
 
-func NewWorkflowRunner(repo github.Repo, a receiverActivities, w Workflow) *WorkflowRunner {
+type dbActivities interface {
+	FetchLatestDeployment(ctx context.Context, request activities.FetchLatestDeploymentRequest) (activities.FetchLatestDeploymentResponse, error)
+	StoreLatestDeployment(ctx context.Context, request activities.StoreLatestDeploymentRequest) error
+}
+
+func NewWorkflowRunner(repo github.Repo, a receiverActivities, db dbActivities, w Workflow) *WorkflowRunner {
 	return &WorkflowRunner{
 		Repo:     repo,
 		Workflow: w,
 		StateReceiver: &StateReceiver{
 			Repo:     repo,
 			Activity: a,
 		},
+		DbActivities: db,
 	}
 }
 
 type WorkflowRunner struct {
 	StateReceiver stateReceiver
 	Repo          github.Repo
 	Workflow      Workflow
+	DbActivities  dbActivities
 }
 
 func (r *WorkflowRunner) Run(ctx workflow.Context, deploymentInfo DeploymentInfo) error {
+	var resp activities.FetchLatestDeploymentResponse
+	err := workflow.ExecuteActivity(ctx, r.DbActivities.FetchLatestDeployment, activities.FetchLatestDeploymentRequest{
+		FullRepositoryName: r.Repo.GetFullName(),
+		RootName:           deploymentInfo.Root.Name,
+	}).Get(ctx, &resp)
+	if err != nil {
+		return errors.Wrap(err, "fetching latest deployment")
+	}
+
+	// TODO: Compare commits to validate revision
```

Comment:
I think this makes more sense doing outside the workflow runner. I'm not sure why this runner needs to care about all this validation part.  Additionally, this struct is already doing a lot.